README.rst

```diff
@@ -133,6 +133,49 @@ input queries before sending them to the API:
 Run ``python -m autoextract --help`` to get description of all supported
 options.
 
+Errors
+~~~~~~
+
+The following errors could happen while making requests:
+
+- Network errors
+- `Request-level errors`_
+    - Authentication failure
+    - Malformed request
+    - Too many queries in request
+    - Request payload size is too large
+- `Query-level errors`_
+    - Downloader errors
+    - Proxy errors
+    - ...
+
+Some errors could be retried while others don't. For example, you could retry
+a request if there's a proxy timeout but you shouldn't retry a request if the
+response is a 404 Not Found.
+
+.. _Request-level errors: https://doc.scrapinghub.com/autoextract.html#request-level
+.. _Query-level errors: https://doc.scrapinghub.com/autoextract.html#query-level
+
+Retries
+~~~~~~~
+
+By default, we will automatically retry Network and Request-level errors.
+You could also enable Query-level errors retries
+by specifying the ``--max-query-error-retries`` argument.
+
+Enable Query-level retries to increase the success rate
+at the cost of more requests being performed
+if you are interested in a higher success rate.
+
+.. code-block::
+
+    python -m autoextract urls.txt --page-type articles --max-query-error-retries 3 --output res.jl
+
+Queries with success will be buffered while failing queries will be retried
+until the max number of retries or a timeout is reached. If it's not possible
```

Comment:
This explanation sounds a bit confusing in the context of the example above. Successful responses are written to a file, not buffered. Technically what you've written is correct though, as CLI uses batch-size 2 by default - so successful responses can be buffered for some time. But how it actually works is a bit confusing without mentioning batches.

What do you think about wording it in a slightly different way? Something along the lines "Failing queries are retried until ...; if not possible, the last available result is written to the output. When batches are used, successful responses may be buffered in memory until all queries in a batch are resolved, because of the limitations of the current implementation."

Reply:
I've updated this part os the readme file.